utils/remoteContentConfig.ts

```diff
@@ -7,145 +8,97 @@ import { parse } from "fp-ts/lib/Json";
-        TE.mapLeft(
-          e =>
-            new Error(
-              `${e.kind}, Remote Content Configuration ServiceId=${serviceId}`
-            )
-        ),
-        TE.chain(
-          TE.fromOption(
-            () =>
-              new Error(
-                `EMPTY_REMOTE_CONTENT_CONFIGURATION, ServiceId=${serviceId}`
-              )
-          )
-        ),
-        TE.chain(remoteContentConfiguration =>
+const RC_CONFIGURATION_REDIS_PREFIX = "RC-CONFIGURATION";
+
+export default class RCConfigurationUtility {
+  constructor(
+    private readonly redisClient: RedisClient,
+    private readonly rcConfigurationModel: RCConfigurationModel,
+    private readonly rcConfigurationCacheTtl: NonNegativeInteger,
+    private readonly serviceToRCConfigurationMap: UlidMapFromString
+  ) {}
+
+  public readonly getOrCacheRCConfigurationWithFallback = (
+    serviceId: NonEmptyString,
+    configurationId?: Ulid
+  ): TE.TaskEither<Error, RetrievedRCConfiguration> =>
+    pipe(
+      configurationId ?? this.serviceToRCConfigurationMap.get(serviceId),
+      Ulid.decode,
+      E.fold(
+        _ => TE.left(new Error(`ConfigurationId is not valid`)),
+        configId =>
           pipe(
-            setWithExpirationTask(
-              redisClient,
-              `REMOTE-CONTENT-CONFIGURATION-${serviceId}`,
-              JSON.stringify(remoteContentConfiguration),
-              remoteContentConfigCacheTtl
-            ),
-            TE.map(() => remoteContentConfiguration),
-            TE.orElse(() => TE.of(remoteContentConfiguration))
+            this.getOrCacheMaybeRCConfigurationById(configId),
+            TE.chain(
+              TE.fromOption(
+                () =>
+                  new Error(
+                    `EMPTY_RC_CONFIGURATION, ConfigurationId=${configId}`
+                  )
+              )
+            )
           )
-        )
       )
-    )
-  );
+    );
 
-export const getOrCacheMaybeRCConfiguration = (
-  redisClient: RedisClient,
-  rCConfigurationModel: RCConfigurationModel,
-  rCConfigurationCacheTtl: NonNegativeInteger,
-  configurationId: Ulid
-): TE.TaskEither<Error, O.Option<RetrievedRCConfiguration>> =>
-  pipe(
-    getTask(redisClient, `REMOTE-CONTENT-CONFIGURATION-${configurationId}`),
-    TE.chain(
-      TE.fromOption(
-        () => new Error("Cannot Get Remote Content Configuration from Redis")
-      )
-    ),
-    TE.chainEitherK(
-      flow(
-        parse,
-        E.mapLeft(
-          () =>
-            new Error(
-              "Cannot parse Remote Content Configuration Json from Redis"
-            )
-        ),
-        E.chain(
-          flow(
-            RetrievedRCConfiguration.decode,
-            E.mapLeft(
-              () =>
-                new Error(
-                  "Cannot decode Remote Content Configuration Json from Redis"
-                )
+  public readonly getOrCacheMaybeRCConfigurationById = (
```

Comment:
Can you add a comment that describes what this method do?

Reply:
done in https://github.com/pagopa/io-functions-app-messages/pull/51/commits/c13e877eb91963770b1631b027fb1ad51972cf10